test/unit/conditions/context.test.ts

```diff
@@ -90,20 +90,16 @@ describe('context parameters', () => {
     };
     const condition = new ContractCondition(conditionObj);
     const conditionExpr = new ConditionExpression(condition);
-    const conditionContext = conditionExpr.buildContext(provider, {}, signer);
```

Comment:
Perhaps you can put this after the `contextRequiresSigner` expect on L93, to prove that the condition context can be build if a signer is passed. Perhaps also add a check that a failure occurs if the signer is not provided...?

Of course if this is tested somewhere else then don't bother.